EIPS/eip-7937.md

```diff
@@ -0,0 +1,145 @@
+---
+eip: 7937
+title: 64-bit mode EVM operations
+description: Multibyte opcodes for 64-bit arithmetic, comparison, bitwise, stack and flow operations in EVM.
+author: Wei Tang (@sorpaas)
+discussions-to: https://ethereum-magicians.org/t/eip-9687-64-bit-mode-evm-operations/23794
+status: Draft
+type: Standards Track
+category: Core
+created: 2025-04-23
+---
+
+## Abstract
+
+This EIP introduces multibyte opcodes prefixed by `C0` for 64-bit arithmetic (`C001`-`C00B`), comparison (`C010`-`C015`), bitwise (`C016`-`C019`), stack (`DUP*` and `SWAP*`) and flow (`C056` and `C057` in "legacy", or `C0E1` and `C0E2` in EOFv1) operations.
+
+## Motivation
+
+Not all computations in EVM can utilize the full 256-bit integer width. It can therefore be beneficial to have a "64-bit mode" to avoid unnecessary cycles. This EIP uses a "prefix" opcode `C0`, essentially forming multibyte opcodes to avoid polluting the EVM opcode space too much.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Prefix opcode behavior
+
+This EIP uses the prefix opcode `C0`, and it only occupies this single EVM opcode space. Upon the interpreter encountering opcode `C0`, it MUST continue to seek the next byte in code. It then executes things in "64-bit mode", based on the second byte, described below. If the execution is successful, then the interpreter MUST increase `PC` by 2 (instead of 1).
+
+If the second byte is not a valid 64-bit mode operation, then the interpreter MUST OOG.
+
+### General 64-bit mode behavior
+
+In 64-bit mode, all operations only work on the lowest 64-bit of each stack value. The first 192-bit is discarded. When a result value is pushed back onto the stack, then it MUST ensures that observable effects will see that the first 192-bit is zero. Note that here it's not necessary for an interpreter to reset the first 192-bit to zero every time -- if the next opcode is still in 64-bit mode, then the first 192-bit is still unobservable. We discuss the full details in the "rationale" section. The interpreter only needs to reproduce the full 256-bit value upon entering non-64-bit mode. If the full computational heavy part can be written in pure 64-bit mode, then this can result in noticable performance gain.
```

Comment:
"lowest" and "first" could be confusing, I think if you change "first 192-bit" to "topmost 192 bits" this is more clear.